pkgs/development/interpreters/lua-5/setup-hook.sh

```diff
@@ -0,0 +1,50 @@
+# set -e
+
+nix_print() {
+    if (( "${NIX_DEBUG:-0}" >= $1 )); then
+        echo "$2"
+    fi
+}
+
+nix_debug() {
+    nix_print 3 "$1"
+}
+
+addToLuaSearchPathWithCustomDelimiter() {
+    # local delimiter="$1"
+    local varName="$1"
+    local pattern="$2"
+    # delete longest match starting from
+    local topDir="${pattern%%\?*}"
+
+    # export only if the folder exists else LUA_PATH grows too big
+    if  [ ! -d "$topDir" ]; then return; fi
+    # check for duplicate
+    if [[ $LUA_PATH = *"$pattern"* ]]; then return; fi
```

Comment:
This check requries that patterns cannot be substrings of each other (which is true; not sure if worth a comment); but shouldn't either the variable to check depend on `varName` or `;$LUA_PATH` just be checked  against `;$topDir/` ?

Reply:
right it should be topDir/$pattern, let me fix it, At the same time, I never had any trouble with duplicate buildInputs so I am tempted to just remove the line.

Reply:
ha no sorry pattern already contains the basefolder so it should be ok as is. I got confused because pattern don't have the same meaning in addToLuaSearchPathWithCustomDelimiter (absolute path) and addToLuaPath (relative path).

Reply:
What I meant is that if `;$LUA_PATH` contains `;$topDir` we know that the bsae directory has been processed already.

Reply:
ok I've dropped the check and renamed pattern to abs(olute)Pattern to make it easier it has a different meaning. No need to troll myself.

Reply:
checking just against topdir would not be enough since several patterns are checked against the same topdir. The first matching pattern would discard others while they may be valid as well.